pkg/aws/eni/instances.go

```diff
@@ -34,7 +34,7 @@ type EC2API interface {
 	AttachNetworkInterface(ctx context.Context, index int32, instanceID, eniID string) (string, error)
 	DeleteNetworkInterface(ctx context.Context, eniID string) error
 	ModifyNetworkInterface(ctx context.Context, eniID, attachmentID string, deleteOnTermination bool) error
-	AssignPrivateIpAddresses(ctx context.Context, eniID string, addresses int32) error
+	AssignPrivateIpAddresses(ctx context.Context, eniID string, addresses int32) ([]string, error)
 	UnassignPrivateIpAddresses(ctx context.Context, eniID string, addresses []string) error
```

Comment:
I also thought about IP release, in particular if `AddIPsToENI` could wrongly re-add an IP that has just been removed. However, I don't think this can happen, since `handleIPRelease`  and `handleIPAllocation` are always called sequentially.

Even without this PR, `handleIPRelease` does not remove the address from the internal book-keeping, so I think we sometimes underestimate the number of available addresses. But this is not changed by this PR at all, fixing `handleIPRelease` to add an equivalent `InstancesManager.RemoveIPsFromENI` can be done in a separate PR.

Reply:
yes. I will raise another PR to handling removing IP and then PD. I'm hoping this PR overall can be a cookie cutter that I can use for the following PRs for removing IP and PD